File: examportal/examfront/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) { }

  // getting quiz questions
  public getQuestionsOfQuiz(qId:any) {
    return this._http.get(`${baseUrl}/question/quiz/all/${qId}`);
  }

  public getQuestionsOfQuizForTest(qId:any) {
    return this._http.get(`${baseUrl}/question/quiz/${qId}`);
  }

  // add question to DB
  public addQuestion(question:any) {
    return this._http.post(`${baseUrl}/question/`, question);
  }

  // delete question 
  public deleteQuestion(qid:any){ 
    return this._http.delete(`${baseUrl}/question/${qid}`);
  }

  // eval quiz
  public evalQuiz(questions:any) {

    return this._http.post(`${baseUrl}/question/eval-quiz`, questions);
  }
}

File: examportal/examserver/examserver/src/main/java/com/exam/service/QuestionService.java
package com.exam.service;

import com.exam.model.exam.Question;
import com.exam.model.exam.Quiz;

import java.util.Set;

public interface QuestionService {

    public Question addQuestion(Question q);

    public Question updateQuestion(Question q);

    public Set<Question> getQuestions();

    public Question getQuestion(Long quesId);

    public Set<Question> getQuestionsOfQuiz(Quiz quiz);

    public void deleteQuestion(Long quesId);

    public Question get(Long quesId);

}
